java/code/src/com/suse/manager/webui/controllers/contentmanagement/mappers/ResponseMappers.java

```diff
@@ -63,6 +63,26 @@ public static ProjectPropertiesResponse mapProjectPropertiesFromDB(ContentProjec
         return properties;
     }
 
+    /**
+     * Map a list of db sources entities to a list of view response beans
+     *
+     * @param sourcesDB the list of db envs
+     * @return the List<EnvironmentResponse> view beans
+     */
+    public static List<ProjectSourceResponse> mapSourcesFromDB(List<ProjectSource> sourcesDB) {
+        return sourcesDB
+                .stream()
+                .map(sourceDb -> {
+                    ProjectSourceResponse projectSourceResponse = new ProjectSourceResponse();
```

Comment:
Maybe you can create a default `Constructor` for this `Class` to receive `id, label, type, state` all in one call?
This could then be written
```
...
.map(sourceDb ->
    new ProjectSourceResponse(
        sourceDb.getId().toString(),
        sourceDb.sourceLabel(),
        sourceDb.sourceType().getLabel(),
        sourceDb.getState().name()
    )
);
...
```

or even better just
```
...
.map(sourceDb -> new ProjectSourceResponse(sourceDb));
...
```
and do the computation in the constructor that receives the `sourceDb` parameter.

Reply:
```
...
.map(sourceDb -> new ProjectSourceResponse(sourceDb));
...
```

tbh the last one, I don't like it. The response bean ProjectSourceResponse class would depend now on a jpa entity bean, it just doesn't seem correct for. I think having somekind of factory pattern is better 

```
.map(sourceDb ->
    new ProjectSourceResponse(
        sourceDb.getId().toString(),
        sourceDb.sourceLabel(),
        sourceDb.sourceType().getLabel(),
        sourceDb.getState().name()
    )
);
```

uh about this one I have mix feelings... 

IMO with sets the code is more readable and flexible, because you can see exactly what you are setting and you avoid the big mess of constructors calling each other with null when you start adding parameters, but on the other end you miss the part where you state "These parameters are mandatory". Both versions have their cons/pros :/

I could apply a builder [pattern](https://dzone.com/articles/what-is-ioc-container-in-spring-what-is-the-spring) with something like:

```java
ProjectSourceResponseBuilder.of(mandatoryName, mandatoryId)
.withParam1(optional1)
.withParam2(optional2)
```

It's the mix of both worlds in a much more readable way, but it seems a bit overkill for this case. Anyway if I change for this one I have to change for all the other response beans in the content-management feature